src/backend/index.js

```diff
@@ -54,6 +54,10 @@ function processFeeds(feeds) {
       const currentFeed = await updateFeed(feed);
       // Add a job to the feed queue to process all of this feed's posts.
       await feedQueue.addFeed(currentFeed);
+      // If failed, set the feed to invalid and save to Redis.
+      feedQueue.on('failed', (job, err) => {
+        currentFeed.setInvalid(err.message);
```

Comment:
When you have a callback in an event handler that is synchronous, and you need to invoke an `async` function, you need to add a `.catch()` so a failed Promise doesn't blow things up.  Also, you shouldn't hold a reference to the `feed` in a closure like this.  Use the `job.data` to recreate a new `Feed` object:

```js
async function invalidateFeed(feedData) {
  const feed = Feed.parse(feedData);
  await feed.setInvalid(feedData.reason || 'unknown reason');
}

feedQueue.on('failed', (job, err) => invalidateFeed(job.data).catch(error => logger.error({ error }, 'Unable to invalidate feed in database.')));
```

NOTE: we'll need to attach the `reason` to the `job.data` before we fail it.  I'm not 100% sure where/how to do this without doing it myself.

Reply:
I tried your above code, it does work... just like you said all of them have a value of 'unknown reason'

Another side effect of it being failed, I'm not sure if its a bug. Bull can't differentiate between feeds failing and posts failing, it adds both to Redis